helix-core/src/main/java/org/apache/helix/monitoring/mbeans/dynamicMBeans/DynamicMBeanProvider.java

```diff
@@ -159,28 +163,30 @@ public synchronized void unregister() {
 
   @Override
   public Object getAttribute(String attribute)
-      throws AttributeNotFoundException, MBeanException, ReflectionException {
+      throws AttributeNotFoundException {
     if (SENSOR_NAME_TAG.equals(attribute)) {
       return getSensorName();
     }
 
-    if (!_attributeMap.containsKey(attribute)) {
-      return null;
+    DynamicMetric metric = _attributeMap.get(attribute);
+    if (metric == null) {
+      throw new AttributeNotFoundException("Attribute[" + attribute + "] is not found.");
```

Comment:
Have you confirmed what will be the output when the get method returns this Exception?
Here's what I want to test:
1. Make the MBeanInfo async with the list. See what is the value and if the fetching value from the MBean server works or not.
2. Fix the async part, then try to fetch the value again. See if the reporting logic still works fine and if we can fetch the right data again.
I just want to be safe.

Reply:
Let me do test it.

Reply:
Test results:
1. MBeanInfo async with the list.
Running 2 threads: thread 1 awaits while it is updating attributes "A" and "B" (not completed); meanwhile, thread 2 updates attributes "C", "D" and completes. Finally, "C", "D" could be fetched correctly, while getting "A" and "B" throws AttributeNotFoundException. thread 1 completes updating. Fetching "A", "B" correctly, and throwing AttributeNotFoundException for fetching "C", "D"
2. Fix async.
Has run a few tests. It works fine and confirms to fetch the data correctly.